### PR TITLE
dump_guest_core: backup config file and modify pre_command

### PR DESCRIPTION
**Pull request by leidwang**
ID:2188907

1.backup the file before the test and restore it after the test
2.make the pre_command modify the config file as expected

Signed-off-by: Leidong Wang <leidwang@redhat.com>

### CODE REVIEW

**Comment by leidwang** [comment]
(1/2) Host_RHEL.m9.u.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9..0.x86_64.io-github-autotest-qemu.dump_guest_core.on.q35: STARTED
 (1/2) Host_RHEL.m9.u.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9..0.x86_64.io-github-autotest-qemu.dump_guest_core.on.q35: PASS (385.18 s)
 (2/2) Host_RHEL.m9.u.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9..0.x86_64.io-github-autotest-qemu.dump_guest_core.off.q35: STARTED
 (2/2) Host_RHEL.m9.u.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9..0.x86_64.io-github-autotest-qemu.dump_guest_core.off.q35: PASS (373.60 s)
RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/avocado/job-results/job-2023-04-23T07.07-4949edb/results.html
JOB TIME   : 760.99 s

**Comment by leidwang** [comment]
Hi @vivianQizhu Could you help to review it?Thanks!

File: qemu/tests/cfg/dump_guest_core.cfg [hunk 1]
```diff
@@ -3,7 +3,7 @@
     only x86_64 ppc64 ppc64le
     virt_test_type = qemu
     pre_command = "echo "<domain>        <type>  <item>  <value>" >> /etc/security/limits.conf"
-    pre_command += "echo "@root          soft    core    unlimited" >> /etc/security/limits.conf"
```
Comment (PaulYuuu) [inline qemu/tests/cfg/dump_guest_core.cfg:6]: These will add 2 lines each time you run the case. Suggest backup the original one or use `sed`.

Comment (leidwang) [reply qemu/tests/cfg/dump_guest_core.cfg:6]: Yes,it is expected.
Okay,will backup it first and restore it after the test.

**Comment by leidwang** [comment]
(1/2) Host_RHEL.m9.u.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9..0.x86_64.io-github-autotest-qemu.dump_guest_core.on.q35: STARTED
 (1/2) Host_RHEL.m9.u.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9..0.x86_64.io-github-autotest-qemu.dump_guest_core.on.q35: PASS (107.48 s)
 (2/2) Host_RHEL.m9.u.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9..0.x86_64.io-github-autotest-qemu.dump_guest_core.off.q35: STARTED
 (2/2) Host_RHEL.m9.u.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9..0.x86_64.io-github-autotest-qemu.dump_guest_core.off.q35: PASS (108.60 s)
RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/avocado/job-results/job-2023-04-25T05.53-c73b548/results.html
JOB TIME   : 218.45 s

**Comment by leidwang** [comment]
Hi @PaulYuuu Could you review it again?Thanks!

**Comment by vivianQizhu** [comment]
Thanks @leidwang @PaulYuuu

**Pull request merged**